### PR TITLE
[`VaSelect`] `VaVirtualScroller` integration, bugfixing

### PR DESCRIPTION
**Pull request by aluarius**
Closes: #740
Closes: #2174 
Closes: #2214 
Closes: #2215 
Closes: #2360 
Closes: #2116 
Partially: #2290

## Description
- [x] Virtual Scrolling option integrated.
- [x] Added slot for option.
- [x] Amount of bugfixes.
- [x] Refactoring & reimplementing.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Comment by rustem-nasyrov** [comment]
Here is extra padding:
image

**Comment by aluarius** [comment]
> Here is extra padding: image

It's made for purpose, because there's no problem with extra padding, but without it sometimes last item can be trunked.

File: packages/ui/src/components/va-select/components/VaSelectOptionList/VaSelectOptionList.vue [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+      return props.options.filter((option: SelectOption) => {
+        const optionText = props.getText(option).toString().toUpperCase()
+        const search = props.search.toUpperCase()
+        return optionText.includes(search)
+      })
+    })
+
+    const optionGroups = computed(() => filteredOptions.value
+      .reduce((groups: Record<string, SelectOption[]>, option) => {
+        if (typeof option !== 'object' || !option.group) {
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-select/components/VaSelectOptionList/VaSelectOptionList.vue:146]:
```suggestion
        if (typeof option !== 'object' || !props.getGroupBy(option)) {
```

**Review by RVitaly1978 — APPROVED** [review]

File: packages/ui/src/components/va-select/components/VaSelectOption/VaSelectOption.vue [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    :style="optionStyle"
+    :aria-selected="isSelected"
+  >
+    <va-icon
+      v-if="optionIcon"
+      size="small"
+      class="va-select-option--icon"
+      :name="optionIcon"
+    />
+    {{ optionText }}
```
[the remark this answers is not in the document]
Comment (aluarius) [reply packages/ui/src/components/va-select/components/VaSelectOption/VaSelectOption.vue:15]: Let's leave it for a complex issue related to the long values (Vasya is needed after all).

File: packages/ui/src/components/va-select/components/VaSelectOptionList/VaSelectOptionList.vue [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+      return props.options.filter((option: SelectOption) => {
+        const optionText = props.getText(option).toString().toUpperCase()
+        const search = props.search.toUpperCase()
+        return optionText.includes(search)
+      })
+    })
+
+    const optionGroups = computed(() => filteredOptions.value
+      .reduce((groups: Record<string, SelectOption[]>, option) => {
+        if (typeof option !== 'object' || !!props.getGroupBy(option)) {
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-select/components/VaSelectOptionList/VaSelectOptionList.vue:146]:
```suggestion
        if (typeof option !== 'object' || !props.getGroupBy(option)) {
```

**Review by RVitaly1978 — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]

**Pull request merged**